doc/user_guide.md

```diff
@@ -13,6 +13,7 @@ The following document is a work in progress and some parts are still missing. T
 - [Configuration](configuration.md)
 - [Debugging failures](debugging.md)
 - [Stateful testing](state.md)
+- [Parallel testing](parallel.md)
```

Comment:
There's a bit of an inflation in what is considered "basic" here :)

Maybe a new section about the stateful testing stuff.